packages/docs/pages/build-integrations/triggers.md

```diff
@@ -0,0 +1,157 @@
+export default [searchCatImages];
+```
+
+:::tip
+If you add new triggers, you need to add them to the `triggers/index.ts` file and export all triggers as an array.
+:::
+
+## Add metadata
+
+Create the `triggers/search-cat-images.ts` file inside of the `thecatapi` folder.
+
+```typescript
+import defineTrigger from '../../../../helpers/define-trigger';
+
+export default defineTrigger({
+  name: 'Search cat images',
+  key: 'searchCatImages',
+  pollInterval: 15,
+  description: 'Triggers when there is a new cat image.',
+
+  async run($) {
+    // TODO: Implement trigger!
+  },
+});
+```
+
+Let's briefly explain what we defined here.
+
+- `name`: The name of the trigger.
+- `key`: The key of the trigger. This is used to identify the trigger in Automatisch.
+- `pollInterval`: The interval in minutes in which the trigger should be polled. Even though we allow to define `pollInterval` field, it's not used in Automatisch at the moment. Currently, the default is 15 minutes and it's not possible to change it.
+- `description`: The description of the trigger.
+- `run`: The function that is executed when the trigger is triggered.
+
+## Implement trigger
+
+:::danger
+
+- Automatisch expects you to push data in **reverse-chronological order** otherwise, the trigger will not work properly.
+- You have to push the `unique identifier` (it can be IDs or any field that can be used to identify the data) as `internalId`. This is used to prevent duplicate data.
+
+:::
+
+Implement the `run` function by adding highlighted lines.
+
+```typescript{6-29}
+import defineTrigger from '../../../../helpers/define-trigger';
+
+export default defineTrigger({
+  // ...
+  async run($) {
+    let page = 0;
+    let response;
+
+    const headers = {
+      'x-api-key': $.auth.data.apiKey,
+    };
+
+    do {
+      let requestPath = `/v1/images/search?page=${page}&limit=10&order=DESC`;
+      response = await $.http.get(requestPath, { headers });
+
+      response.data.forEach((image: IJSONObject) => {
+        const dataItem = {
+          raw: image,
+          meta: {
+            internalId: image.id,
+          },
+        };
+
+        $.pushTriggerItem(dataItem);
+      });
+
+      page += 1;
+    } while (response.data.length >= 10);
+  },
+});
+```
+
+We are using the `$.http` object to make HTTP requests. Our API is paginated, so we are making requests until we get less than 10 items, which means the last page.
+
+We do not have to return anything from the `run` function. We are using the `$.pushTriggerItem` function to push the data to Automatisch. $.pushTriggerItem accepts an object with the following fields:
+
+- `raw`: The raw data that you want to push to Automatisch.
+- `meta`: The metadata of the data. It has to have the `internalId` field.
+
+:::tip
+
+`$.pushTriggerItem` is smart enough to understand if the data is already pushed to Automatisch or not. If the data is already pushed, it will stop the trigger, otherwise, it will continue to fetch new data.
```

Comment:
For the curious contributors, it may be beneficial to mention that pushTriggerItem is scoped by the flow. Meaning that the same piece of data is not considered "already processed/pushed" in different flows. What do you think, @farukaydin?